csharp/src/Google.Protobuf/JsonFormatter.cs

```diff
@@ -170,6 +170,132 @@ public void Format(IMessage message, TextWriter writer)
         }
 
         /// <summary>
+        /// Formats the specified field of the message as JSON.
+        /// </summary>
+        /// <param name="message">The message to format.</param>
+        /// <param name="field">The field in the message to format.</param>
+        /// <returns>The formatted field.</returns>
+        public string FormatFieldValue(IMessage message, FieldDescriptor field)
```

Comment:
This can format any type of field. Single, repeated and map. It's converted to the usual representation.